core/planner.c

```diff
@@ -1080,6 +1080,13 @@ bool plan(struct deco_state *ds, struct diveplan *diveplan, struct dive *dive, i
 		diveplan->eff_gflow = lrint(100.0 * (regressiona() * first_stop_depth + regressionb()));
 	}
 
+	for (int i = 0; i < MAX_CYLINDERS; i++)
+		if (cylinder_nodata(&dive->cylinder[i])) {
+			// Switch to an empty air cylinder for breathing air at the surface
+			current_cylinder = i;
+			break;
+		}
```

Comment:
obvious question... what happens if the user has all available cylinders defined and none of them are air?
I know, sounds like a corner case, but I also know that those are the things that tend to bite us.

Reply:
I thought about this as well. First of all, this should not happen(™). Remember the discussion when we increased MAX_CYLIDERS to its current value and some senior people complained that "this is far too big and occupies too much memory"? But we increased it anyway to have room when merging dives into different dive computers of a single dive (and in that case, it's not a problem as they would share the "surface air" gas). But we could increase MAX_CYLINDERS by one...

That said, what happens now if we run out of gases to have an additional surface air gas is to continue to breath the last deco gas. I admit, this could be confusing but it is failing gravely.

Possible alternatives: when hitting MAX_CYLINDERS, forcing the extra time to 0, effectively turning this feature off. That would be a bit more code but probably a nicer failure mode. Or I could stop modelling this surface segment breathing surface air by an additional cylinder and making this segment "more special" with the implicit meaning of breathing air. But I guess that would introduce "if (segment.type == SPECIAL)" cases in many many places.

Reply:
You mean it's failing gracefully? I agree, that's not a terrible failure mode. And indeed, the case is unlikely to happen.
And on the overall tradeoff of this vs extra code for special handling... let's keep it simple.